foundation/aissemble-foundation-core-python/src/policy_manager/policy/policy.py

```diff
@@ -49,9 +50,49 @@ class ConfiguredRule(BaseModel):
     invoke the rule will be responsible for looking up the related class.
     """
 
+    __logger = LogManager.get_instance().get_logger("ConfiguredRule")
     className: str
     configurations: Optional[Dict[str, Any]]
-    targetConfigurations: Optional[ConfiguredTarget]
+    configuredTargets: Optional[List[ConfiguredTarget]] = []
+
+    @property
+    def targetConfigurations(self) -> ConfiguredTarget:
+        """
+        This attribute is deprecated and should not be used. ConfiguredTarget are now represented as
+        a List of ConfiguredTarget's instead of a single ConfiguredTarget attribute `targetConfigurations`.
+        This attribute is replaced by `configuredTargets`.
+        """
+        ConfiguredRule.__logger.warn(
+            "Detected use of deprecated attribute 'targetConfigurations'. Existing "
+            + "usage should be moved to the new attribute 'configuredTargets'."
+        )
+        return self.configuredTargets[0] if len(self.configuredTargets) > 0 else None
+
+    def set_deprecated_targetConfigurations(self, new_value: ConfiguredTarget):
```

Comment:
Q/A: Do we need to support the setter? I'd have thought this was a read-only property that's determined by the definition JSON file.

Reply:
In theory we probably don't since I think these would only be set by the abstract policy manager which does it through the new property. However the java implementation has a now deprecated setter for it, so can't hurt to maintain parity here.